test/state/requests.cpp

```diff
@@ -42,7 +45,8 @@ Requests collect_deposit_requests(std::span<const TransactionReceipt> receipts)
     {
         for (const auto& log : receipt.logs)
         {
-            if (log.addr == DEPOSIT_CONTRACT_ADDRESS)
+            assert(log.addr != DEPOSIT_CONTRACT_ADDRESS || !log.topics.empty());
+            if (log.addr == DEPOSIT_CONTRACT_ADDRESS && log.topics[0] == DEPOSIT_TOPIC)
```

Comment:
We usually don't do this, but I'm thinking about also checking if the `topics[0]` exists and if the length of the data is enough. But we can wait for the https://github.com/ethereum/EIPs/pull/9460.